### PR TITLE
Accessibility tests setup

### PR DESCRIPTION
**Pull request by youriwims**
Using `cypress-axe` + `react-axe` + potentially `eslint-plugin-jsx`

Also see https://github.com/mozilla/foundation.mozilla.org/tree/axe-core-ci-testing

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -6,15 +6,19 @@
     "autoprefix": "postcss --use autoprefixer -o network-api/networkapi/frontend/_css/main.compiled.css network-api/networkapi/frontend/_css/main.compiled.css",
     "build:common": "shx rm -rf network-api/networkapi/frontend && shx mkdir -p network-api/networkapi/frontend/_js && run-p build:images build:sass && npm run build:contribute",
     "build:contribute": "shx cp contribute.json network-api/networkapi/frontend/",
+    "build:dev": "npm run build:common && npm run build:js:dev",
```
Comment (Pomax) [inline package.json]: `npm-run-all` is already installed, so:
```suggestion
    "build:dev": "run-s build:common build:js:dev",
```

File: package.json [hunk 2]
```diff
@@ -30,7 +34,7 @@
     "test:scss:styleguide:color": "stylelint \"source/sass/**/*.scss\" \"source/js/**/*.scss\" \"!source/sass/**/_colors.scss\" --syntax scss --config .stylelintrc-colors.js",
     "test:css": "stylelint \"network-api/networkapi/{,!(frontend)/**/}*.css\" --syntax scss",
     "test": "run-s test:** build",
-    "watch": "npm run build && run-p watch:**",
+    "watch": "npm run build:dev && run-p watch:**",
```
Comment (Pomax) [inline package.json]: ideally this would be `run-s build:dev watch-all` with a separate task `"watch-all": "run-p watch:*`

File: package.json [hunk 3]
```diff
@@ -6,15 +6,19 @@
     "autoprefix": "postcss --use autoprefixer -o network-api/networkapi/frontend/_css/main.compiled.css network-api/networkapi/frontend/_css/main.compiled.css",
     "build:common": "shx rm -rf network-api/networkapi/frontend && shx mkdir -p network-api/networkapi/frontend/_js && run-p build:images build:sass && npm run build:contribute",
     "build:contribute": "shx cp contribute.json network-api/networkapi/frontend/",
+    "build:dev": "npm run build:common && npm run build:js:dev",
+    "build:dev:a11y": "npm run cypress:install && npm run cy:open",
```
Comment (Pomax) [inline package.json]:
```suggestion
    "build:dev:a11y": "run-s cypress:install cy:open",
```

**Comment by Pomax** [comment]
@cadecairos this works, sort of, but it does install a _lot_ of stuff and it takes quite a long time to run, so I don't know if we actually want this as part of the default review app CI cycle... It feels like making this a developer pass checkbox ("did you verify that accessibility tests pass") as part of filing the PR might be better, especially as fixing errors is a matter of visiting the page in question with the Axe web extension installed (Chrome/Firefox) and running an analysis with dev tools for the page in question.

**Comment by cadecairos** [comment]
@Pomax Yeah, I don't think that we should be downloading and unzipping Cypress by default. Maybe we can re-organize the Docker containers and compose file so that there's a way to get and test with Cypress, but by default it's the same process as it was before.

**Comment by cadecairos** [comment]
This was resolved

**Comment by cadecairos** [comment]
haha wrong tab. CARRY ON

**Comment by cadecairos** [comment]
So one thing to note, the current setup is incompatible with npm commands running in a docker container.

**Comment by Pomax** [comment]
Closing this one in favour of #4197

**Comment by Pomax** [comment]
I'll leave the branch hanging around until we land the rebased PR, though.

**Pull request closed without merging**